src/shell/shell_history.cpp

```diff
@@ -81,7 +81,9 @@ ShellHistory::ShellHistory() : path(get_shell_history_path())
 	}
 	auto history_file = std::ifstream(path);
 	if (!history_file) {
-		path.clear();
+		if (std::filesystem::exists(path)) {
+			path.clear();
+		}
 		return;
 	}
```

Comment:
>  When moving the shell history to its own class, I opted to have the path cleared if the ifstream failed to open. This was because if the file couldn't be read but could still be written to (ie. strange file permissions), it would delete whatever was in the history file.

I don't follow this logic.  `std::ofstream` truncates by default so we are always erasing and replacing the contents of the file.  If you want to handle invalid permissions, maybe log a warning but I don't see what clearing the path accomplishes.

Reply:
That's fair.  I would still suggest logging a warning when this happens instead of silently failing.